handlers/photo-haptics-audio-handler/src/server.ts

```diff
@@ -0,0 +1,327 @@
+    const preObjDet = preprocessors["ca.mcgill.a11y.image.preprocessor.objectDetection"]
+    const preGroupData = preprocessors["ca.mcgill.a11y.image.preprocessor.grouping"];
+
+    if (!preObjDet && !preSemSeg && !preGroupData) {
+        console.debug("No preprocessor data available: can't render!");
+        const response = utils.generateEmptyResponse(req.body["request_uuid"]);
+
+        if (ajv.validate("https://image.a11y.mcgill.ca/handler-response.schema.json", response)) {
+            res.json(response);
+        } else {
+            console.error("Failed to generate a valid empty response!");
+            console.error(ajv.errors);
+            res.status(500).json(ajv.errors);
+        }
+        return;
+    }
+
+    // *******************************************************
+    // Check for renderer availability
+    // *******************************************************
+    const hasText = req.body["renderers"].includes("ca.mcgill.a11y.image.renderer.Text");
+    const hasAudioHaptic = req.body["renderers"].includes("ca.mcgill.a11y.image.renderer.PhotoAudioHaptics");
+    if (!hasAudioHaptic && !hasText) {
+        console.warn("Segment audio-haptic renderers not supported!");
+        const response = utils.generateEmptyResponse(req.body["request_uuid"]);
+        if (ajv.validate("https://image.a11y.mcgill.ca/handler-response.schema.json", response)) {
+            res.json(response);
+        } else {
+            console.error("Failed to generate a valid empty response!");
+            console.error(ajv.errors);
+            res.status(500).json(ajv.errors);
+        }
+        return;
+    }
+
+    // *******************************************************
+    // Audio TTS
+    // *******************************************************
+    // Begin forming text...
+    // This is variable depending on which preprocessor data is available.
+    const ttsData: utils.TTSSegment[] = [];
+    ttsData.push({"value": utils.generateIntro(preSecondCat), "type": "text"});
+    if (preSemSeg) {
+        // Use all segments returned for now.
+        // Filtering may be helpful later.
+        ttsData.push(...utils.generateSemSeg(preSemSeg));
+        if (preObjDet && preGroupData) {
+            ttsData.push({"value": "It also", "type": "text"});
+        }
+    }
+    if (preObjDet && preGroupData) {
+        ttsData.push(...utils.generateObjDet(preObjDet, preGroupData));
+    }
+
+        // Concatenate adjacent text entries
+        for (let i = 0; i < ttsData.length - 1; i++) {
+            if (ttsData[i].type === "text" && ttsData[i+1].type === "text") {
+                ttsData[i].value += " " + ttsData[i+1].value;
+                ttsData.splice(i+1, 1);
+            }
+        }
+
+    // Generate rendering title
+    const renderingTitle = utils.renderingTitle(preSemSeg, preObjDet, preGroupData);
+
+    // Construct text (if requested)
+    if (hasText) {

```

Comment:
This seems to just duplicate the text rendering of photo-audio-handler, which would likely already be running. Is there a reason to create a text output for this handler?